File: backend-springboot/src/main/java/com/mentis/usuario/dto/PreguntaDTO.java
package com.mentis.usuario.dto;

import java.util.List;

/** Pregunta enviada al candidato. NUNCA incluye la respuesta correcta. */
public record PreguntaDTO(
    Long id,
    Integer orden,
    String tipo,
    String categoria,
    String enunciado,
    List<String> opciones,
    Integer puntos,
    String respuestaCandidato,
    boolean respondida
) {}

File: mentis-android/app/src/main/java/com/mentis/app/data/remote/dto/PreguntaDto.kt
package com.mentis.app.data.remote.dto

import com.google.gson.annotations.SerializedName

data class PreguntaDto(
    @SerializedName("id") val id: Int,
    @SerializedName("orden") val orden: Int,
    @SerializedName("tipo") val tipo: String, // "multiple" | "abierta"
    @SerializedName("categoria") val categoria: String,
    @SerializedName("enunciado") val enunciado: String,
    @SerializedName("opciones") val opciones: List<String>,
    @SerializedName("puntos") val puntos: Int,
    @SerializedName("respuestaCandidato") val respuestaCandidato: String?,
    @SerializedName("respondida") val respondida: Boolean
)